src/algokit/cli/tasks/analyze.py

```diff
@@ -0,0 +1,230 @@
+    ),
+)
+@click.option(
+    "-e",
+    "--exclude",
+    "detectors_to_exclude",
+    multiple=True,
+    default=[],
+    type=click.STRING,
+    help="Exclude specific vulnerabilities from the analysis. Supports multiple exclusions in a single run.",
+)
+def analyze(  # noqa: PLR0913
+    *,
+    input_paths: tuple[Path],
+    recursive: bool,
+    force: bool,
+    diff_only: bool,
+    output_path: Path | None,
+    detectors_to_exclude: list[str],
+) -> None:
+    """
+    Analyze TEAL programs for common vulnerabilities using Tealer.
+    """
+
+    detectors_to_exclude = sorted(set(detectors_to_exclude))
+    input_files = get_input_files(input_paths=input_paths, recursive=recursive)
+
+    if not force:
+        click.confirm(
+            click.style(
+                "Warning: This task uses `tealer` to suggest improvements for your TEAL programs, "
+                "but remember to always test your smart contracts code and follow modern "
+                "software engineering practices. Do you understand?",
+                fg="yellow",
+            ),
+            default=True,
+            abort=True,
+        )
+
+    reports = {}
+    duplicate_files: dict[str, int] = {}
+    prepare_artifacts_folders(output_path)
+    total_files = len(input_files)
+    for index in range(total_files):
+        cur_file = input_files[index]
+        file = cur_file.resolve()
+
+        if has_template_vars(file):
+            click.secho(
+                f"Warning: Skipping {file} due to template variables. Substitute them before scanning.",
+                err=True,
+                fg="yellow",
+            )
+            continue
+
+        filename = generate_report_filename(file, duplicate_files)
+
+        # If diff_only is enabled, store the report in the snapshots folder otherwise store it in the reports folder
+        # If a custom output path is provided, store the report in the specified path
+        report_output_root = output_path or TEALER_SNAPSHOTS_ROOT
+        report_output_path = report_output_root / filename
+
+        command = generate_tealer_command(cur_file, report_output_path, detectors_to_exclude)
+        old_report = load_tealer_report(str(report_output_path)) if report_output_path.exists() and diff_only else None
+        try:
+            run_with_animation(run_tealer, f"Analyzing {index + 1} out of {total_files} files", command)
+
+            if diff_only and old_report:
+                has_diff = has_baseline_diff(
+                    cur_file=cur_file, report_output_path=report_output_path, old_report=old_report
+                )
+                if has_diff:
+                    report_output_path.write_text(json.dumps(old_report.model_dump(by_alias=True), indent=2))
+                    raise click.exceptions.Exit(1)
+
+            reports[str(report_output_path.absolute())] = json.load(report_output_path.open())
+        except Exception as e:
```

Comment:
If an exception is thrown and diff_only is true, we probably want to revert the report back to old_report, as tealer may have written the current report to disc.